spec/spec_helper.rb

```diff
@@ -67,3 +68,10 @@ def unindent
 end
 
 Dir[File.expand_path('../support/**/*.rb', __FILE__)].each {|f| require f }
+
+RSpec.configure do |config|
+  config.before(:each) do 
+    stub_request(:get,'https://raw.githubusercontent.com/anshuman23/ruBayes/master/test/sample.csv').
+      to_return(status: 200, body: "a,b,c,d,e\n4.8,3.4,1.9,0.2,Iris-setosa\n5.0,3.0,1.6,0.2,Iris-setosa\n5.0,3.4,1.6,0.4,Iris-setosa\n5.2,3.5,1.5,0.2,Iris-setosa\n6.6,3.0,4.4,1.4,Iris-versicolor\n6.8,2.8,4.8,1.4,Iris-versicolor\n6.7,3.0,5.0,1.7,Iris-versicolor\n6.0,2.9,4.5,1.5,Iris-versicolor\n6.4,3.1,5.5,1.8,Iris-virginica\n6.0,3.0,4.8,1.8,Iris-virginica\n6.9,3.1,5.4,2.1,Iris-virginica\n6.7,3.1,5.6,2.4,Iris-virginica\n")
+  end
```

Comment:
This stub implies _most_ of your tests needs it, which is not true. The stub should be in io_spec.rb, near the actual test. And contents of "stabbed" csv should not be pasted in code, but stored in `spec/fixtures` folder (or rather use one of existing there fixtures).